File: pset2/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// pset2/caesar.c
// Implement a program that encrypts messages using Caesar’s cipher
// c is computed as c[i] = (p[i] + k) % 26

// Program must accept a single argument, a non-negative integer k
int main(int argc, string argv[])
{
    if (argc != 2)
    {
        // Error message if no argument or too many arguments
        printf("Usage: ./caesar k\n");
        return 1;
    }

    // Convert string to int
    int k = atoi(argv[1]);
    // printf("%d", k);

    // Prompt user for a plaintext string
    printf("plaintext:  ");
    string p = get_string();

    printf("ciphertext: ");

    // Loop through the plaintext string
    for (int i = 0, n = strlen(p); i < n; i++)
    {
        // Check that the character is a letter
        if ((p[i] >= 'A' && p[i] <= 'Z') || (p[i] >= 'a' && p[i] <= 'z'))
        {
            // Determine if the character is uppercase
            if (p[i] >= 'A' && p[i] <= 'Z')
            {
                // Calculate ciher for uppercase letter
                printf("%c", ((p[i] + k - 'A') % 26) + 'A');
            }

            // Determine if the character is lowercase
            if (p[i] >= 'a' && p[i] <= 'z')
            {
                // Calculate ciher for lowercase letter
                printf("%c", ((p[i] + k - 'a') % 26) + 'a');
            }
        }
        else
        {
            // Print characters that are not letters
            printf("%c", p[i]);
        }
    }

    // After outputting ciphertext, you should print a newline
    printf("\n");

    // Your program should then exit by returning 0 from main
    return 0;
}

File: pset6/caesar/caesar.py
import cs50
import sys
from cs50 import get_string

# program must accept a single command-line argument, a non-negative integer. Let’s call it k
if len(sys.argv[1:]) < 1 or len(sys.argv[1:]) > 1:
    print("usage: ./caesar.py k")
    exit(1)

# k must be a non-negative integer (e.g., 1). No need to check that it’s indeed numeric.
k = int(sys.argv[1])

# prompt the user for a string of plaintext
p = ""
while p == "":
    p = get_string("plaintext: ")

# loop through plaintext string and encrypt the letters
# program must preserve case
print("ciphertext: ", end='')
for c in p:
    if(c.isalpha()):
        if(c.isupper()):
            print(chr(((ord(c) + k - ord('A')) % 26) + ord('A')), end='')
        else:
            print(chr(((ord(c) + k - ord('a')) % 26) + ord('a')), end='')
    else:
        print(c, end='')

# After outputting ciphertext, you should print a newline.
print()